octobot_trading/personal_data/portfolios/portfolio_value_holder.py

```diff
@@ -261,21 +291,74 @@ def _try_get_value_of_currency(self, currency, quantity, raise_error):
+            )
             return quantity / self._get_last_price_data(reversed_symbol)
-        except decimal.DivisionByZero:
-            pass
-        except KeyError:
+        except (KeyError, decimal.DivisionByZero):
             pass
-        raise errors.MissingPriceDataError(f"no price data to evaluate {current_currency} price in {target_currency}")
+        raise errors.MissingPriceDataError(
+            f"no price data to evaluate {current_currency} price in {target_currency}"
+        )
+
+    def try_convert_currency_value_using_multiple_pairs(self, currency, quantity) -> decimal.Decimal:
+        # settlement_asset needs to be handled to add support for futures
+
+        # try with two pairs
+        # for example:
+        # currency: ETH - ref market: USDT
+        # ETH/USDT is not available. ETH/BTC and BTC/USDT are available though.
+        # first convert ETH -> BTC and then BTC -> USDT
+        #               | bridge part 1     | bridge part 2
+
+        # look into available symbols to find pair bridges
+        part_1_base = currency
+        part_2_quote = self.portfolio_manager.reference_market
+        for bridge_part_1_symbol in self.last_prices_by_trading_pair:
+            parsed_bridge_part_1_symbol = symbol_util.parse_symbol(bridge_part_1_symbol)
+            # part 1: check if bridge_part_1_symbol can be used as part 1 of the bridge
+            if part_1_base != parsed_bridge_part_1_symbol.base:
+                continue
+            part_1_quote = parsed_bridge_part_1_symbol.quote
+            try:
+                bridge_part_1_value = self.convert_currency_value_using_last_prices(
+                    quantity, part_1_base, part_1_quote
+                )
+            except errors.MissingPriceDataError:
+                # first pair might not be initialized or is not available at all
+                # can't be used as a bridge part 1, try with other pairs
+                continue
+            # check that bridge_part_1_value is really set
+            if not bridge_part_1_value:
+                continue
+            # part 2: bridge part 1 is found and valued, try to get a compatible second part
+            try:
+                bridge_part_2_value = self.convert_currency_value_using_last_prices(
+                    constants.ONE,
+                    part_1_quote,
+                    part_2_quote,
+                )
+                if bridge_part_2_value:
+                    if currency in self.missing_currency_data_in_exchange:
+                        self.missing_currency_data_in_exchange.remove(currency)
+                    return bridge_part_1_value * bridge_part_2_value
+            except errors.MissingPriceDataError:
+                # conversion pairs might not be initialized or is not available at all
+                pass  # continue with other pairs
+        # no bridge found
+        return None
 
```

Comment:
Should we log something here?

Reply:
not here, in the caller there is an appropriate error handling